apps/aecore/src/pow_sha256.erl

```diff
@@ -0,0 +1,104 @@
+%%%=============================================================================
+%%% @copyright 2017, Aeternity Anstalt
+%%% @doc
+%%%    Module dealing with SHA256 Proof of Work calculation and hash generation
+%%% @end
+%%%=============================================================================
+-module(pow_sha256).
+
+-export([generate/3,
+         recalculate_difficulty/3]).
+
+-ifdef(TEST).
+-compile(export_all).
+-endif.
+
+-include("sha256.hrl").
+
+-define(NONCE_RANGE, 1000000000000000000000000).
+
+%% Integers encrypted in scientific notation:
+%%   256 * <base-2 exponent> + the most significant byte (i.e., the significand)
+-type sci_int() :: integer().
+-type pow_result() :: integer() | {'error', 'generation_count_exhausted'}.
```

Comment:
I guess you may prefer `{ok, integer()}` to `integer()` here so to ease matching.